# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/LoadedScript.h
# Commit: c11cffaded9c
# Full Hash: c11cffaded9cfed53b04e5c7875278e9a9e55396
# Author: Yulia Startsev <ystartsev@mozilla.com>
# Date: 2022-02-18 21:52:29
# Regressor Bug: 1742437
# File Overlap Count: 5
# Description:
#   Bug 1742437 - re-scope all generic module/script loader files under js/loader; r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138133
# ==============================================================================

diff -r 0527d227c6b6 -r c11cffaded9c js/loader/LoadedScript.h
--- a/js/loader/LoadedScript.h	Fri Feb 18 12:45:52 2022 +0000
+++ b/js/loader/LoadedScript.h	Fri Feb 18 12:45:53 2022 +0000
@@ -4,8 +4,8 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-#ifndef mozilla_dom_LoadedScript_h
-#define mozilla_dom_LoadedScript_h
+#ifndef js_loader_LoadedScript_h
+#define js_loader_LoadedScript_h
 
 #include "nsCOMPtr.h"
 #include "nsCycleCollectionParticipant.h"
@@ -14,11 +14,7 @@
 
 class nsIURI;
 
-namespace mozilla {
-namespace dom {
-
-class Element;
-class ScriptLoader;
+namespace JS::loader {
 
 void HostAddRefTopLevelScript(const JS::Value& aPrivate);
 void HostReleaseTopLevelScript(const JS::Value& aPrivate);
@@ -31,11 +27,11 @@
   ScriptKind mKind;
   RefPtr<ScriptFetchOptions> mFetchOptions;
   nsCOMPtr<nsIURI> mBaseURL;
-  nsCOMPtr<Element> mElement;
+  nsCOMPtr<mozilla::dom::Element> mElement;
 
  protected:
   LoadedScript(ScriptKind aKind, ScriptFetchOptions* aFetchOptions,
-               nsIURI* aBaseURL, Element* aElement);
+               nsIURI* aBaseURL, mozilla::dom::Element* aElement);
 
   virtual ~LoadedScript();
 
@@ -54,7 +50,7 @@
   ScriptFetchOptions* GetFetchOptions() const { return mFetchOptions; }
 
   // Used by the Debugger to get the associated Script Element
-  Element* GetScriptElement() const { return mElement; }
+  mozilla::dom::Element* GetScriptElement() const { return mElement; }
 
   nsIURI* BaseURL() const { return mBaseURL; }
 
@@ -66,7 +62,7 @@
 
  public:
   ClassicScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
-                Element* aElement);
+                mozilla::dom::Element* aElement);
 };
 
 class EventScript final : public LoadedScript {
@@ -74,7 +70,7 @@
 
  public:
   EventScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
-              Element* aElement);
+              mozilla::dom::Element* aElement);
 };
 
 // A single module script. May be used to satisfy multiple load requests.
@@ -93,7 +89,7 @@
                                                          LoadedScript)
 
   ModuleScript(ScriptFetchOptions* aFetchOptions, nsIURI* aBaseURL,
-               Element* aElement);
+               mozilla::dom::Element* aElement);
 
   void SetModuleRecord(JS::Handle<JSObject*> aModuleRecord);
   void SetParseError(const JS::Value& aError);
@@ -123,7 +119,6 @@
   return static_cast<ModuleScript*>(this);
 }
 
-}  // namespace dom
-}  // namespace mozilla
+}  // namespace JS::loader
 
-#endif  // mozilla_dom_LoadedScript_h
+#endif  // js_loader_LoadedScript_h